### DISCUSSION TITLE
Are there serial adapters which have a resistor behind RX?

### DISCUSSION THREAD

**Discussion by SquashSquishSquaw** [SpenceKonde/megaTinyCore#459] [Q&A] [answered]
Hi,

in the Serial Adapter to UPDI Programmer Guide, there's this recommendation:

> The better kind are what you find on ebay when you search for "CH340 6pin" - you want the black ones with the voltage switch.

I think the one I've bought is one of these, and I've installed a 1N60P diode between RX and TX.

The guide also recommends to put a 470 Ohm resistor between the pins Adapter(RX) and Target(UPDI). If I'm not mistaken however, my serial adapter seems to have not only a 1 k resistor right behind the Adapter(TX) pin, but also a second 1 k resistor right behind the Adapter(RX) pin. Is that something very unusual to have? If I really have that, would I still need the 470 Ohm resistor between target and adapter?

Many thanks
Jenny

**Comment by SpenceKonde** [comment, marked as answer]
I've never seen it before, burt I believe it. Care to show me a link to product listing or at least photo, since that is rather rare? 

Trying make sure it doesn't blow iup when someone connects unlevel shifted RS232 to it matybe? 

A resistor between RX pin on board, amnd RX of serial chip should have basically no effect if you have tehschittjy diode 

I clearly need to a better guide because lost of people are having trouble following it having trouble with the results  when it works flawlessly for me every time. Beyond the fact that I don't haver time to reweritethat longass guide, it I do not think that I can do any better if I were to try again, The resistor on the target side is recommended for the sole purpose that it permits programming with the cheapo Arduiiono Nanop via jtag2updi, without having to work a resistor in there. All SerialUPDI needs is that there be a resistor somewhere the serial chip's TX amd tje target's UPDI pin  assuming there's a schottky diode in use  and that total resistance isn't too high. Or if no diode, the tiotal resistance needs to be within a very narrow margin around= 4.7k

**Reply by SquashSquishSquaw** [reply]
Many thanks for your very helpful answer, Spence!

I think the manual for your software package is great the way it is. It did take me a short while to figure out the structure of the document "Step-by-step guide to turn serial adapter or uno/nano/pro mini into a UPDI programmer", but all necessary information is there, and it's explained in a very understandable way.

MY ATtinys haven't arrived yet so I can't test it right now, but I'm quite confident that it will work, provided that the noname 1N60P is suitable for the job. If I understand correctly, the 470 Ohm resistor between the pins Adapter(RX) and Target(UPDI) also has the job to prevent device destruction if I connect something the wrong way. Apart from that I assume that I wouldn't need it then, since I don't plan to use jtag2updi.


The adapter which I bought, can be found on Ebay:

https://www.ebay.com/itm/381998739870
https://i.ebayimg.com/images/g/KmoAAOSw2gxYw8CN/s-l1600.jpg

On the image can be seen a line of 4 components right next and parallel to the line of six IO pins, from left to right an LED, and then three resistors, 1 2 and 3.

If I measure from TX pin to right side of resistor 2, I measure 1 kOhm, and if I measure from RX pin to right side of resistor 3, I also measure 1 kOhm. The right side of resistor 3 connects to a little through-hole right next to it (about where the lower left corner of the pushbutton is), which – now on the backside of the PCB – is connected to another through-hole, directly under the 16-pin CH340G IC. Unfortunately I can't see what's going on there because the IC covers everything, but the through-hole is very close to pin 3 on the IC, which I believe is RXD. So it seems to me that between CH340G(Pin3=RXD) and Adapter(RXD-Pin), there is 1 kOhm.

If you'd like some more pictures I could try to take some.

Best regards, have a nice weekend
Jenny

**Reply by SpenceKonde** [reply]
Those are the least flawed of the commercial adapters that have a voltage switch IMO. (am working on bringing to market some that actually do it right... but until then, what you're using is the best of the,. Add just a schottky diode and you should be good. (as an aside, that's  also the model I ship as the optional serial adapter on my Tindie store,

They do everything right, except that they don't actually switch the voltage that the chip runs at, so it's just that resistor and the protection diodes keeping current injection to safe levels when using in 3.3v mode (there is one that also switches chip Vdd but it has horrible design defects and was clearly the work of an inexperienced designer.). The ones I;m working  on will have micro USB, so you don't need extension cables, just common micro USB cables, the power LED will change color depending on which voltage it's set to and they don't have a mysterious button of unknown function either.... Prototypes are working werll and only planned change is that the pins on the end need to be flipped so they match what they get plugged into

**Reply by SquashSquishSquaw** [reply]
Thank you Spence! :-)

> and they don't have a mysterious button of unknown function either

It's good to hear that I'm not the only one wondering what that one might be good for. :-D On one of the pictures it says, "Button used for STC MCU series", but I have no clue what it does.

Best regards
Jenny
